fractal_server/app/runner/_slurm/executor.py

```diff
@@ -0,0 +1,228 @@
+# This adapts clusterfutures <https://github.com/sampsyo/clusterfutures>
+# Original Copyright
+# Copyright 2021 Adrian Sampson <asampson@cs.washington.edu>
+# License: MIT
+#
+# Copyright 2022 Jacopo Nepsolo <jacopo.nespolo@exact-lab.it>
+import shlex
+import subprocess  # nosec
+import sys
+from concurrent import futures
+from pathlib import Path
+from typing import Any
+from typing import Callable
+from typing import List
+from typing import Optional
+
+import cloudpickle
+from cfut import RemoteException
+from cfut import SlurmExecutor  # type: ignore
+from cfut.util import random_string
+
+from ....config import get_settings
+from ....syringe import Inject
+from ....utils import set_logger
+
+
+def get_slurm_script_dir() -> Path:
+    settings = Inject(get_settings)
+    script_dir = settings.RUNNER_ROOT_DIR  # type: ignore
+    return script_dir  # type: ignore
+
+
+def get_stdout_filename(arg: str = "%j") -> Path:
+    return get_slurm_script_dir() / f"slurmpy.stdout.{arg}.log"
+
+
+def get_in_filename(arg) -> Path:
+    return get_slurm_script_dir() / f"cfut.in.{arg}.pickle"
+
+
+def get_out_filename(arg) -> Path:
+    return get_slurm_script_dir() / f"cfut.out.{arg}.pickle"
+
+
+def write_batch_script(
+    sbatch_script: str, script_dir: Optional[Path] = None
+) -> Path:
+    """
+    Write batch script
+
+    Returns:
+        batch_script_path:
+            The path to the batch script
+    """
+    if not script_dir:
+        script_dir = get_slurm_script_dir()
+
+    batch_script_path = script_dir / f"_temp_{random_string()}.sh"
```

Comment:
As above, I think we should move this to a specific folder #284.
Not urgent, can be done in future "cleanup" PRs.

Reply:
will solve #284 separately.

Also, it is not just here that might need changes. Also the sbatch script sets output and error for the job.